### DISCUSSION TITLE
Snippets into Data Table Support

### DISCUSSION THREAD

**Discussion by Vasperous** [facelessuser/pymdown-extensions#1779] [Q&A] [answered]
Hello! I am looking to include a table within a table using Material for Mkdocs.  What I have found is that doing this in markdown doesn't appear to possible so I have started to build the tables, copy the HTML and then paste into the high level table.

```
**Test 1**    | **Test 2**
:-----------: | :------------------
Testing1      | Testing
Testing2      | Testing <div class="md-typeset__scrollwrap"><div class="md-typeset__table"><table> <thead> <tr> <th align="center"><strong>Test 1</strong></th> <th align="left"><strong>Test 2</strong></th> </tr> </thead> <tbody> <tr> <td align="center">Testing1</td> <td align="left">Testing</td> </tr> <tr> <td align="center">Testing2</td> <td align="left">Testing</td> </tr> </tbody> </table></div></div>
```

As a result I get the following which looks as desired:

![image](https://user-images.githubusercontent.com/8703632/183523348-0d5aa0ca-5b6c-4dc0-80b6-1e2f462d20f9.png)

I use snippet a bunch for my logs/examples which have significant code so keeping them in separate files is extremely helpful to keep my other pages clean (thank you!).  So when I couldn't put one table inside another I thought that snippets might be able to help me out.  For example:

```
**Test 1**    | **Test 2**
:-----------: | :------------------
Testing1      | Testing
Testing2      | Testing  --8<-- "test.md" 
```

Unfortunately what I am getting is the following:

![image](https://user-images.githubusercontent.com/8703632/183524182-22177ddd-3842-42ce-a302-4107d5b7931b.png)

Any ideas on how to get the snippets to insert the contents of the `test.md` into a data table?  Thank you!

**Comment by facelessuser** [comment, marked as answer]
I'm afraid I don't fully understand what you are trying to achieve. The syntax you are using makes no sense to me. What is in your snippet?

**Reply by facelessuser** [reply]
Are you trying to load an entire table into a single table cell? If so, Snippets only if it is the only thing on a single line. Even if it worked the way you show in your example, it would create invalid Markdown syntax. Snippets does a find and replace of the Snippet syntax with the referenced file. If the final text is invalid Markdown, it won't parse as the Markdown structure you intend.

**Reply by Vasperous** [reply]
Thank you for the reply!

> Are you trying to load an entire table into a single table cell?

Yes, the HTML that is working is a single line is working:

```
**Test 1**    | **Test 2**
:-----------: | :------------------
Testing1      | Testing
Testing2      | Testing <div class="md-typeset__scrollwrap"><div class="md-typeset__table"><table> <thead> <tr> <th align="center"><strong>Test 1</strong></th> <th align="left"><strong>Test 2</strong></th> </tr> </thead> <tbody> <tr> <td align="center">Testing1</td> <td align="left">Testing</td> </tr> <tr> <td align="center">Testing2</td> <td align="left">Testing</td> </tr> </tbody> </table></div></div>
```

I guess I was hoping that snippets would render/convert the contents of the *.md file into HTML etc. first and then insert it the documentation where I am using the `--8<--`.  So if that isn't happening then I agree that it won't structure or work the way I intended.

Could you see a situation where if I were to first convert the *.md files in my snippets folder to HTML first, remove new lines etc. and then do the replace that this would work?

**Reply by facelessuser** [reply]
If you render the whole thing first, you lose context. I use Snippets to attach various references to the end of every page. If it was rendered first, it would defeat the purpose. It is done as a preprocessor step.

You can certainly pre convert content and then include it with Snippets, but, unfortunately, the documentation is also pretty clear about that it must be on a line by itself:

> The important thing is that the notation must reside on a line(s) by itself, and the path must be quoted in the case of the single line notation.

**Reply by Vasperous** [reply]
Completely understood, I guess I need to figure out a way to convert the markdown to HTML and move it to one line before the snippets begins the processing.

**Reply by Vasperous** [reply]
Actually, I guess snippets won't work in a table, meaning, this is not supported:

```
**Test 1**    | **Test 2**
:-----------: | :------------------
Testing1      | Testing
Testing2      | Testing  --8<-- "test.md" 
```

So in short, is it fair to say holistically that snippets doesn't work in a data table?

**Reply by facelessuser** [reply]
Well, not exactly, you could do something like this:

Snippet:

```
Testing3   |  Testing
Testing4   |  Testing
```

Then do this to add more rows.

```
**Test 1**    | **Test 2**
:-----------: | :------------------
Testing1      | Testing
Testing2      | Testing 
--8<-- "test.md" 
```

You just can't do what you want. Snippets works on inserting file content as a block, not inline content.

**Reply by Vasperous** [reply]
Thank you for the explanation, makes sense =)
